Modern RabbitMQ

binarin opened the pull request:
###### Motivation for this change

This series of commits upgrades RabbitMQ to latest stable version and modernizes corresponding NixOS module. Most important changes are:
- Proper integration with systemd: socket-activated epmd, startup notifications, sane defaults
- Using new configuration file format, which is more automation friendly

Parts of Erlang infrastructure were also touched to make socket-activation real.

Every commit message also contains more detailed description, together with relevant links to RabbitMQ and Erlang/OTP repositories.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

grahamc commented:
@GrahamcOfBorg build rabbitmq

GrahamcOfBorg reviewed (commented):
Failure for system: aarch64-linux

```
error: attribute 'rabbitmq' in selection path 'rabbitmq' not found
```

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-linux

```
error: attribute ‘rabbitmq’ in selection path ‘rabbitmq’ not found
```

grahamc commented:
...
@GrahamcOfBorg build rabbitmq_server

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-darwin

```
error: attribute 'rabbitmq' in selection path 'rabbitmq' not found
```

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-darwin

```
while evaluating the attribute '__impureHostDeps' of the derivation 'erlang-19.3.6.4' at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/stdenv/generic/make-derivation.nix:148:11:
while evaluating 'unique' at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/lib/lists.nix:455:12, called from /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/stdenv/generic/make-derivation.nix:137:13:
while evaluating 'concatMap' at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/lib/lists.nix:102:18, called from /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/stdenv/generic/make-derivation.nix:137:25:
while evaluating anonymous function at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/stdenv/generic/make-derivation.nix:137:40, called from undefined position:
while evaluating 'chooseDevOutputs' at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/lib/attrsets.nix:460:22, called from undefined position:
while evaluating 'optional' at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/lib/lists.nix:198:20, called from /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/development/interpreters/erlang/generic-builder.nix:57:8:
while evaluating 'callPackageWith' at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/lib/customisation.nix:113:35, called from /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/top-level/all-packages.nix:13416:13:
while evaluating 'makeOverridable' at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/lib/customisation.nix:72:24, called from /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/lib/customisation.nix:117:8:
while evaluating anonymous function at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/os-specific/linux/systemd/default.nix:1:1, called from /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/lib/customisation.nix:74:12:
assertion failed at /Users/graham/nix-borg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/grahamc-ndnd/pkgs/os-specific/linux/systemd/default.nix:9:1
```

GrahamcOfBorg reviewed (commented):
Success for system: x86_64-linux

```
gzipping man pages under /nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2/share/man/
strip is /nix/store/mdyy001q67hiks0g24ra53z7ckm4jfr4-binutils-2.28.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2/sbin 
patching script interpreter paths in /nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2
/nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2/escript/rabbitmq-diagnostics: interpreter directive changed from " /usr/bin/env escript" to "/nix/store/2sd8mmc3kah6q8fm2jvi1838q5f4yc9g-erlang-19.3.6.4/bin/escript"
/nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2/escript/rabbitmq-plugins: interpreter directive changed from " /usr/bin/env escript" to "/nix/store/2sd8mmc3kah6q8fm2jvi1838q5f4yc9g-erlang-19.3.6.4/bin/escript"
/nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2/escript/rabbitmqctl: interpreter directive changed from " /usr/bin/env escript" to "/nix/store/2sd8mmc3kah6q8fm2jvi1838q5f4yc9g-erlang-19.3.6.4/bin/escript"
checking for references to /tmp/nix-build-rabbitmq-server-3.7.2.drv-0 in /nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2...
moving /nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2/sbin/* to /nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2/bin
/nix/store/8bb2njibgd65c8vqd560jxisl535jazx-rabbitmq-server-3.7.2
```

GrahamcOfBorg reviewed (commented):
Failure for system: aarch64-linux

```
escript /build/source/bootstrap/lib/erl_docgen/priv/bin/specs_gen.escript -I../../include -I../../src -o../specs/ ../../src/gen/wxRadioBox.erl
escript /build/source/bootstrap/lib/erl_docgen/priv/bin/specs_gen.escript -I../../include -I../../src -o../specs/ ../../src/gen/wxRadioButton.erl
escript /build/source/bootstrap/lib/erl_docgen/priv/bin/specs_gen.escript -I../../include -I../../src -o../specs/ ../../src/gen/wxRegion.erl
escript /build/source/bootstrap/lib/erl_docgen/priv/bin/specs_gen.escript -I../../include -I../../src -o../specs/ ../../src/gen/wxSashEvent.erl
escript /build/source/bootstrap/lib/erl_docgen/priv/bin/specs_gen.escript -I../../include -I../../src -o../specs/ ../../src/gen/wxSashLayoutWindow.erl
building of '/nix/store/r3066978pg0h9rbadmf9af7kxpi9v70p-erlang-19.3.6.4.drv' timed out after 3200 seconds
cannot build derivation '/nix/store/m345vq87z12r36w5dn8zy1r0cg5v3gcc-rebar-2.5.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/12nkp7wlz066kv8xcqaph1mijq25krmk-elixir-1.5.3.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/99611m0arba044f2f079m0m9niyi0v3f-rabbitmq-server-3.7.2.drv': 2 dependencies couldn't be built
error: build of '/nix/store/99611m0arba044f2f079m0m9niyi0v3f-rabbitmq-server-3.7.2.drv' failed
```

adisbladis commented:
Tests are passing on `x86_64-linux` 👍 

I'm very much in favour of this PR but I want @grahamc to give the final OK. :)

binarin commented:
Bumped rabbit to recently released 3.7.3, moved elixir version selection to all-packages.nix as suggested.

Profpatsch commented:
Since `3.6` [is going to be supported until May 2018](https://groups.google.com/forum/#!msg/rabbitmq-users/kXkI-f3pgEw/UFowJIK4BQAJ), I request that an up-to-date `3.6.x` is kept in nixpkgs.

Profpatsch reviewed (changes requested):
Since `3.6` [is going to be supported until May 2018](https://groups.google.com/forum/#!msg/rabbitmq-users/kXkI-f3pgEw/UFowJIK4BQAJ), I request that an up-to-date `3.6.x` is kept in nixpkgs.

Here is a patch to 3.6.14: https://gitlab.techcultivation.org/Profpatsch/nixpkgs-techcultivation/commit/38afdc9923f3b6aa601023b599a34dcc4af58f96

binarin commented:
@Profpatsch I can easily do this for package itself, but I don't think I can do this in a clean way for NixOS module. If you are OK with that, how'd you prefer the new and the old package named in all-packages?

Profpatsch commented:
> but I don't think I can do this in a clean way for NixOS module.

I’d to it like this:

For the packages:
```
let
  majorVersion = 6;
in … {
  version = "${toString majorVersion}.x.x";
  …
  passthrough = { inherit majorVersion; };
}
```

And then use `cfg.package.majorVersion >= 7` to switch in the module.

> how'd you prefer the new and the old package named in all-packages?

They should probably be named
```
rabbitmq-server = rabbitmq-server_3_6;
rabbitmq-server_3_6 = …
rabbitmq-server_3_7 = …
```

Notice that I changed the (historical) `_` to `-`, that needs to be documented in the release notes under `nixos/doc/manual/release-notes/`.

Profpatsch commented:
fyi: I’ve created a PR to update to `3.6.15`, with various improvements to the package. Some of them are probably valid for `3.7` as well and should be merged into the `3.7` derivation as well.

I intend to merge the `3.6` update first, since we should support the latest release of that major release (see above).

peterhoeg commented:
While we're at it, can we use ```erlang_nox``` instead of regular ```erlang``` by default when building rabbitmq? We've been using that for the last 6 months or so without any problems.

Profpatsch commented:
I’m for it, I [already disable wx for our rabbitmq docker image](https://gitlab.techcultivation.org/sangha/sangha-deployment/blob/bd04f830d0bb46e4152a8710fcae7aa26e163f61/containers/rabbitmq.nix#L26).

As you can see in that source file, the wx beam files are put in nonetheless, so `erlang_nox` should probably also throw them out if possible. Rabbitmq itself depends on `coreutils` because of one command used in a wrapper script, that can probably be patched out as well.

Profpatsch commented:
The deprecation for the `3.6` series is coming nearer (May 1st), so we should get this into `nixpkgs`.

We should still keep `3.6` around (though with a deprecation warning) and remove it with the next release (`18.09`).

peterhoeg commented on nixos/modules/services/networking/epmd.nix:
```diff
@@ -0,0 +1,60 @@
+
+      serviceConfig = {
+        User = "nobody";
+        Group = "nogroup";
```
We shouldn't be using nobody/nogroup - those are reserved for NFS. Instead either run with ```DynamicUser = true;``` (preferred) if possible or alternatively create a user/group for this.

peterhoeg commented on nixos/modules/services/networking/epmd.nix:
```diff
@@ -0,0 +1,60 @@
+        ExecStart = "${cfg.package}/bin/epmd -systemd";
+        Type = "notify";
+        StandardOutput = "journal";
+        StandardError = "journal";
```
```journal``` is default anyway, so you don't have to set those.

binarin commented:
I updated this patch without re-introducing version-selection logic described above, it's already to late.

This depends on:
- https://github.com/NixOS/nixpkgs/pull/43817
- https://github.com/NixOS/nixpkgs/pull/43812

`ps` with `systemd` support is now required because of https://github.com/rabbitmq/rabbitmq-server/blob/7c941aff893c44355e3e892a43b80d20de171ac3/src/rabbit.erl#L415

binarin committed:
rabbitmq-server: 3.6.10 -> 3.7.8

- New dependency on 'getconf' binary for
  https://github.com/rabbitmq/rabbitmq-common/blob/3aa619e9ef81cfe6db0c776e6cd829ce234a7881/src/vm_memory_monitor.erl#L448
- New dependency on 'socat' for systemd notifications
  https://github.com/rabbitmq/rabbitmq-server/blob/4a3ee3a336bc82f2e56ec5d2292c80458ccc2974/src/rabbit.erl#L361
- elixir_1_6 for a new 'rabbitmqctl' tool
- Replace patching with providing custom PATH, as we already have some
  other things here
- Renamed package in all-packages.nix from a legacy spelling

binarin committed:
rabbitmq module: modernize after package upgrade

- Use socket-activated epmd - that way there won't be any trouble when
  more than one erlang system is used within a single host.
- Use new automation-friendly configuration file format
- Use systemd notifications instead of buggy 'rabbitmqctl wait' for
  confirming successful server startup.
  'wait' bug: https://github.com/rabbitmq/rabbitmq-server/issues/463
- Use 'rabbitmqctl shutdown' instead of 'stop', because it's not
  pid-file based
- Use sane systemd unit defaults from RabbitMQ repo:
  https://github.com/rabbitmq/rabbitmq-server/blob/master/docs/rabbitmq-server.service.example
- Support for external plugins

binarin commented:
@Profpatsch All dependencies are in `master` now, and I've rebased/updated the PR itself.

Profpatsch commented:
@GrahamcOfBorg build rabbitmq-server

GrahamcOfBorg commented:
<!--REQUEST_ID=629c7965-ca93-4d19-b43b-0bd46d22f358-->
Unexpected error: command failed with exit code 1 on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34265&attempt_id=d39c19f0-4994-43e1-9ad8-fd07c225d722)

Attempted: rabbitmq-server

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/qjk2h301faawvbnhv5azhyiaw6vrqalg-builder.sh: fork: retry: Resource temporarily unavailable
/nix/store/qjk2h301faawvbnhv5azhyiaw6vrqalg-builder.sh: fork: retry: Resource temporarily unavailable
copying path '/nix/store/vsi0n294sn2g8r6sndjk3c7fr9pq2r5z-erlang-20.3.8' from 'https://cache.nixos.org'...
/nix/store/qjk2h301faawvbnhv5azhyiaw6vrqalg-builder.sh: fork: retry: Resource temporarily unavailable
/nix/store/qjk2h301faawvbnhv5azhyiaw6vrqalg-builder.sh: fork: retry: Resource temporarily unavailable
copying path '/nix/store/yx1sv0lqzglr4hl0i1fv76nfba285vki-elixir-1.6.6' from 'https://cache.nixos.org'...
/nix/store/qjk2h301faawvbnhv5azhyiaw6vrqalg-builder.sh: fork: Resource temporarily unavailable
builder for '/nix/store/nn45bfclc6fh4ya15axaw6ifijcrq4dq-rabbitmq-server-3.7.8.tar.xz.drv' failed with exit code 254
cannot build derivation '/nix/store/0w5f7jh3xg5jyixf4q52zgsdzxs5lhpx-rabbitmq-server-3.7.8.drv': 1 dependencies couldn't be built
error: build of '/nix/store/0w5f7jh3xg5jyixf4q52zgsdzxs5lhpx-rabbitmq-server-3.7.8.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=629c7965-ca93-4d19-b43b-0bd46d22f358-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34265&attempt_id=30f15deb-0a82-4da2-ad2f-7f966714208f)

Attempted: rabbitmq-server

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/g30cvk3ay8hv56wxfmjk08hn7gzi6q2h-rabbitmq-server-3.7.8-man
gzipping man pages under /nix/store/g30cvk3ay8hv56wxfmjk08hn7gzi6q2h-rabbitmq-server-3.7.8-man/share/man/
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/g30cvk3ay8hv56wxfmjk08hn7gzi6q2h-rabbitmq-server-3.7.8-man
checking for references to /build in /nix/store/g30cvk3ay8hv56wxfmjk08hn7gzi6q2h-rabbitmq-server-3.7.8-man...
shrinking RPATHs of ELF executables and libraries in /nix/store/76dc97my11vin2d7ymlbnl0s7j74s75s-rabbitmq-server-3.7.8-doc
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/76dc97my11vin2d7ymlbnl0s7j74s75s-rabbitmq-server-3.7.8-doc
checking for references to /build in /nix/store/76dc97my11vin2d7ymlbnl0s7j74s75s-rabbitmq-server-3.7.8-doc...
/nix/store/v35gyq73wpkn8gi7v0lgb759hsskjvla-rabbitmq-server-3.7.8
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=629c7965-ca93-4d19-b43b-0bd46d22f358-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34265&attempt_id=ced7c07a-b60e-4ae9-accb-ee64bd089d8a)

Attempted: rabbitmq-server

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/hhhz83blfp8l2n45y8a5sipzvazj017a-erlang-20.3.8/lib/erlang/erts-9.3.3/bin/erl: interpreter directive changed from "/bin/sh" to "/nix/store/fqm2x6kiay1q4vg7pqp4wp17bdijlyc3-bash-4.4-p23/bin/sh"
/nix/store/hhhz83blfp8l2n45y8a5sipzvazj017a-erlang-20.3.8/lib/erlang/erts-9.3.3/bin/start_erl.src: interpreter directive changed from "/bin/sh" to "/nix/store/fqm2x6kiay1q4vg7pqp4wp17bdijlyc3-bash-4.4-p23/bin/sh"
/nix/store/hhhz83blfp8l2n45y8a5sipzvazj017a-erlang-20.3.8/lib/erlang/erts-9.3.3/bin/start.src: interpreter directive changed from "/bin/sh" to "/nix/store/fqm2x6kiay1q4vg7pqp4wp17bdijlyc3-bash-4.4-p23/bin/sh"
/nix/store/hhhz83blfp8l2n45y8a5sipzvazj017a-erlang-20.3.8/lib/erlang/erts-9.3.3/bin/erl.src: interpreter directive changed from "/bin/sh" to "/nix/store/fqm2x6kiay1q4vg7pqp4wp17bdijlyc3-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/hhhz83blfp8l2n45y8a5sipzvazj017a-erlang-20.3.8...
wrong ELF type
wrong ELF type
wrong ELF type
cannot build derivation '/nix/store/1kgsv2m4k4zkfsbpab20kbjvaly7bjca-rabbitmq-server-3.7.8.drv': 1 dependencies couldn't be built
error: build of '/nix/store/1kgsv2m4k4zkfsbpab20kbjvaly7bjca-rabbitmq-server-3.7.8.drv' failed
```
</p></details>

Profpatsch commented:
I think the darwin failure is unrelated. Let’s try again.

@GrahamcOfBorg build rabbitmq-server

GrahamcOfBorg commented:
<!--REQUEST_ID=21c2866b-158f-4fa9-97f9-2081edf3c7cd-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34265&attempt_id=5185ad3d-4cf4-4df4-8d1e-74af417d8dd6)

Attempted: rabbitmq-server

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/g30cvk3ay8hv56wxfmjk08hn7gzi6q2h-rabbitmq-server-3.7.8-man
gzipping man pages under /nix/store/g30cvk3ay8hv56wxfmjk08hn7gzi6q2h-rabbitmq-server-3.7.8-man/share/man/
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/g30cvk3ay8hv56wxfmjk08hn7gzi6q2h-rabbitmq-server-3.7.8-man
checking for references to /build in /nix/store/g30cvk3ay8hv56wxfmjk08hn7gzi6q2h-rabbitmq-server-3.7.8-man...
shrinking RPATHs of ELF executables and libraries in /nix/store/76dc97my11vin2d7ymlbnl0s7j74s75s-rabbitmq-server-3.7.8-doc
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/76dc97my11vin2d7ymlbnl0s7j74s75s-rabbitmq-server-3.7.8-doc
checking for references to /build in /nix/store/76dc97my11vin2d7ymlbnl0s7j74s75s-rabbitmq-server-3.7.8-doc...
/nix/store/v35gyq73wpkn8gi7v0lgb759hsskjvla-rabbitmq-server-3.7.8
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=21c2866b-158f-4fa9-97f9-2081edf3c7cd-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34265&attempt_id=dba043b9-006f-40c5-99db-ec8eaff9f59b)

Attempted: rabbitmq-server

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/pqfpya2v3m44abrkdp101mjhmwi8blv3-rabbitmq-server-3.7.8/escript/rabbitmqctl: interpreter directive changed from " /usr/bin/env escript" to "/nix/store/8imgai416j7ih0yn3ma3mdvc9b8f3a6a-erlang-20.3.8/bin/escript"
/nix/store/pqfpya2v3m44abrkdp101mjhmwi8blv3-rabbitmq-server-3.7.8/escript/rabbitmq-diagnostics: interpreter directive changed from " /usr/bin/env escript" to "/nix/store/8imgai416j7ih0yn3ma3mdvc9b8f3a6a-erlang-20.3.8/bin/escript"
/nix/store/pqfpya2v3m44abrkdp101mjhmwi8blv3-rabbitmq-server-3.7.8/escript/rabbitmq-plugins: interpreter directive changed from " /usr/bin/env escript" to "/nix/store/8imgai416j7ih0yn3ma3mdvc9b8f3a6a-erlang-20.3.8/bin/escript"
moving /nix/store/pqfpya2v3m44abrkdp101mjhmwi8blv3-rabbitmq-server-3.7.8/sbin/* to /nix/store/pqfpya2v3m44abrkdp101mjhmwi8blv3-rabbitmq-server-3.7.8/bin
gzipping man pages under /nix/store/47rkpi7s4hp60qa9rd2gn4n5njmg3mv9-rabbitmq-server-3.7.8-man/share/man/
strip is /nix/store/4w56qihlrddav67p7d1vy5qkyayaqw11-cctools-binutils-darwin/bin/strip
patching script interpreter paths in /nix/store/47rkpi7s4hp60qa9rd2gn4n5njmg3mv9-rabbitmq-server-3.7.8-man
strip is /nix/store/4w56qihlrddav67p7d1vy5qkyayaqw11-cctools-binutils-darwin/bin/strip
patching script interpreter paths in /nix/store/kllsiq1c0kzhkn7yy9gw23h9h0y8h2ax-rabbitmq-server-3.7.8-doc
/nix/store/pqfpya2v3m44abrkdp101mjhmwi8blv3-rabbitmq-server-3.7.8
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=21c2866b-158f-4fa9-97f9-2081edf3c7cd-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.34265&attempt_id=e31623b8-123a-44a8-a7aa-e9844761de8a)

Attempted: rabbitmq-server

<details><summary>Partial log (click to expand)</summary><p>

```
gzipping man pages under /nix/store/9gbvnjvrbqdrzd5vgrvaqj46wi694lmb-elixir-1.6.6/share/man/
strip is /nix/store/y4ymnvgxygpq05h03kyzbj572zmh6zla-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/9gbvnjvrbqdrzd5vgrvaqj46wi694lmb-elixir-1.6.6/lib  /nix/store/9gbvnjvrbqdrzd5vgrvaqj46wi694lmb-elixir-1.6.6/bin
patching script interpreter paths in /nix/store/9gbvnjvrbqdrzd5vgrvaqj46wi694lmb-elixir-1.6.6
/nix/store/9gbvnjvrbqdrzd5vgrvaqj46wi694lmb-elixir-1.6.6/lib/elixir/bin/iex: interpreter directive changed from "/bin/sh" to "/nix/store/fqm2x6kiay1q4vg7pqp4wp17bdijlyc3-bash-4.4-p23/bin/sh"
/nix/store/9gbvnjvrbqdrzd5vgrvaqj46wi694lmb-elixir-1.6.6/lib/elixir/bin/elixirc: interpreter directive changed from "/bin/sh" to "/nix/store/fqm2x6kiay1q4vg7pqp4wp17bdijlyc3-bash-4.4-p23/bin/sh"
/nix/store/9gbvnjvrbqdrzd5vgrvaqj46wi694lmb-elixir-1.6.6/lib/elixir/bin/elixir: interpreter directive changed from "/bin/sh" to "/nix/store/fqm2x6kiay1q4vg7pqp4wp17bdijlyc3-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/9gbvnjvrbqdrzd5vgrvaqj46wi694lmb-elixir-1.6.6...
cannot build derivation '/nix/store/1kgsv2m4k4zkfsbpab20kbjvaly7bjca-rabbitmq-server-3.7.8.drv': 1 dependencies couldn't be built
error: build of '/nix/store/1kgsv2m4k4zkfsbpab20kbjvaly7bjca-rabbitmq-server-3.7.8.drv' failed
```
</p></details>
